edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs

```diff
@@ -110,31 +110,55 @@ public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int>
         }
 
-        public Task<SystemInfo> GetSystemInfo()
+        public async Task<SystemInfo> GetSystemInfo()
         {
-            // Comment this out to unblock testing.
-            // V1NodeList k8SNodes = await this.client.ListNodeAsync();
-            // string osType = string.Empty;
-            // string arch = string.Empty;
-            // string version = string.Empty;
-            // if (k8SNodes.Items != null)
-            // {
-            //     V1Node firstNode = k8SNodes.Items.FirstOrDefault();
-            //     if (firstNode?.Status?.NodeInfo != null)
-            //     {
-            //         osType = firstNode.Status.NodeInfo.OperatingSystem;
-            //         arch = firstNode.Status.NodeInfo.Architecture;
-            //         version = firstNode.Status.NodeInfo.OsImage;
-            //     }
-            //     else
-            //     {
-            //         Events.NullNodeInfoResponse(firstNode?.Metadata?.Name ?? "UNKNOWN");
-            //     }
-            // }
+            V1NodeList k8sNodes = await this.client.ListNodeAsync();
+            string accumulatedArch = string.Empty;
+
+            // Use "Kubernetes" and the current API version as analogy to device OS/version
             string osType = "Kubernetes";
-            string arch = "Kubernetes";
-            string version = "Kubernetes";
-            return Task.FromResult(new SystemInfo(osType, arch, version));
+            string version = k8sNodes.ApiVersion;
+
+            // Enumerate each node and count the unique architectures
+            var archList = new List<(string arch, int count)>();
+            foreach (V1Node node in k8sNodes.Items)
+            {
+                if (node?.Status?.NodeInfo != null)
+                {
+                    bool haveSeen = false;
+
+                    // Check to see if we've already seen this architecture
+                    for (int i = 0; i < archList.Count; i++)
+                    {
+                        if (archList[i].arch == node.Status.NodeInfo.Architecture)
+                        {
+                            // Just increase the node count
+                            archList[i] = (archList[i].arch, archList[i].count + 1);
+                            haveSeen = true;
+                            break;
+                        }
+                    }
+
+                    // Add an entry if this is a new arch
+                    if (!haveSeen)
+                    {
+                        archList.Add((node.Status.NodeInfo.Architecture, 1));
+                    }
+                }
+                else
+                {
+                    Events.NullNodeInfoResponse(node?.Metadata?.Name ?? "UNKNOWN");
+                    break;
+                }
+            }
+
+            // Construct the accumulated architecture string
+            foreach (var (arch, count) in archList)
+            {
+                accumulatedArch += arch + "(" + count + ")\t";
```

Comment:
@darobs Should we preserve the same structure as Docker mode or we can make it different? For example like this: 
```
{
  OperatingSystemType: "Kubernetes",
  Version: "1.12",
  Architectures: [
    { Name: "x86", NodesCount: 10 },
    { Name: "arm64", NodesCount: 1}
  ]
}
```